web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -75,14 +90,37 @@
         @RequestParam(defaultValue = "ASC") Direction direction
     ) {
         String sort = sortBy == null ? null : sortBy.getOriginalValue();
-        
+        List<String> studyIds = null;
+        if (usingAuth()) {
+            /*
+             If using auth, filter the list of samples returned using the list of study ids the
+             user has access to. If the user has access to no studies, the endpoint should not 403,
+             but instead return an empty list.
+            */
+            studyIds = studyService
+                .getAllStudies(
+                    null,
+                    projection.name(),
+                    PagingConstants.MAX_PAGE_SIZE,
+                    0,
+                    null,
+                    direction.name()
+                )
+                .stream()
+                .map(CancerStudy::getCancerStudyIdentifier)
+                .collect(Collectors.toList());
+        }
+
         if (projection == Projection.META) {
             HttpHeaders httpHeaders = new HttpHeaders();
-            httpHeaders.add(HeaderKeyConstants.TOTAL_COUNT, sampleService.getMetaSamples(keyword).getTotalCount().toString());
+            httpHeaders.add(
+                HeaderKeyConstants.TOTAL_COUNT,
+                sampleService.getMetaSamples(keyword, studyIds).getTotalCount().toString()
+            );
```

Comment:
This is the continuation of line 116. I think our general style is to close function calls on the last line containing arguments rather than pushing the closing parenthesis for the call to sit on its own line. So in this case, the pattern I have seen most often would be to have line 118 end with ".toString());"
If this closing parenthesis is to remain on its own line, the google style requires indentation.
I think we could have a project-wide discussion on style .. I think you made the point that having fewer connecting points in an argument list allows changes which reduce the number of changed lines .. which would be nice. In that case, possibly we should move to starting each line in an argument continuation line with a comma, rather than ending with a comma, and also move the closing parenthesis to a line on its own. But if we adopt that style we should apply it uniformly throughout our codebase.

Reply:
For now I'll change it to reflect the status quo.